classes/State.php

```diff
@@ -54,7 +54,7 @@ class StateCore extends ObjectModel
             'id_country' => ['type' => self::TYPE_INT, 'validate' => 'isUnsignedId', 'required' => true],
             'id_zone' => ['type' => self::TYPE_INT, 'validate' => 'isUnsignedId', 'required' => true],
             'iso_code' => ['type' => self::TYPE_STRING, 'validate' => 'isStateIsoCode', 'required' => true, 'size' => 7],
-            'name' => ['type' => self::TYPE_STRING, 'validate' => 'isGenericName', 'required' => true, 'size' => 32],
+            'name' => ['type' => self::TYPE_STRING, 'validate' => 'isGenericName', 'required' => true, 'size' => 64],
```

Comment:
```suggestion
            'name' => ['type' => self::TYPE_STRING, 'validate' => 'isGenericName', 'required' => true, 'size' => 255],
```
I don't see any reason to set it lower than 255

Reply:
If so, we should modify the ps_state table structure (modify the .sql file somewhere inside install folder), and I think it may cause conflicts with the current 1.7.x installations

As right now, it is set to 64 chars in the table structure which is large enough (gotta find a state name larger than that)

Reply:
Using an existing States database from all around the world, brings `Southern Nations, Nationalities, and Peoples' Region` from Ethiopia with 52 characters as the longest state name in English language. Finding a state name using 64 characters or more may be possible in other languages, but for convenience in every day use, I believe we will always find an abbreviation shorter than 64 chars.

> The Southern Nations, Nationalities, and Peoples' Region (often abbreviated as SNNPR; Amharic: የደቡብ ብሔር ብሔረሰቦችና ሕዝቦች ክልል, romanized: Yädäbub Bḥer Bḥeräsäbočna Hzboč Kllə) [Wikipedia](https://en.wikipedia.org/wiki/Southern_Nations,_Nationalities,_and_Peoples%27_Region)
>> 6 chars abbreviated, 25 chars in Amharic and 36 chars romanized

![image](https://user-images.githubusercontent.com/36200981/147970180-3a480e1b-6bd4-4fcd-bcef-09c7d1259db5.png)

So bumpping it to 255 characters seems a little bit too excessive and a waste of resources on limited or shared hosts, specially for memory tables before results [see discussion in StackExchange](https://dba.stackexchange.com/questions/76469/mysql-varchar-length-and-performance). Also 255 chars will be excessive to show as an option in a form list.

But wdyt?

Reply:
Initially I was thinking about homogeneity. Most "medium sized text" fields are usually stored as varchar 255. Having one field that does not fit this rule makes it an exception and any exception make the design a little harder to learn/maintainer for new users.

But looking at prestashop database schema I see a lot of different varchar sizes :p so 64 seems OK to me.